### PR TITLE
chore: move variants to feature library under @spartacus/product with separate entry point

### LINKED ISSUE #10989: Move product variants to feature library and depracate storefrontlib entries
It's a good time to move basic product variants to a separate library.
Variants should be placed under @spartacus/product/variants, as sub-feature module with separate entry-point.
Additionally, provide deprecation data for the existing ones.

### PR DESCRIPTION
**Pull request by psmul**
Closes: #10989

### CODE REVIEW

File: feature-libs/product/package.json [hunk 1]
```diff
@@ -11,11 +11,27 @@
...
+    "@spartacus/core": "3.1.0-next.0",
+    "@spartacus/storefront": "3.1.0-next.0",
+    "bootstrap": "^4.3.1",
+    "rxjs": "^6.6.0"
+  },
   "publishConfig": {
     "access": "public"
-  }
+  },
+  "schematics": "./collection.json"
```
Comment (dunqan) [inline feature-libs/product/package.json]: We would like to disable schematics for now, until the new version will be ready (by just removing this line)

File: feature-libs/product/public_api.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 /*
  * Public API Surface of product
  */
-export {};
+
+export * from '@spartacus/product/variants';
```
Comment (dunqan) [inline feature-libs/product/public_api.ts]: We don't want to reexport sub entry point in the root one anymore (so we like to keep `export {}` )

File: projects/storefrontlib/src/cms-components/product/product-variants/guards/product-variant.guard.ts [hunk 3]
```diff
@@ -15,6 +15,10 @@ import {
 import { Observable, of } from 'rxjs';
 import { filter, map, switchMap, take } from 'rxjs/operators';
 
+/**
+ * @deprecated since 3.1
```
Comment (dunqan) [inline projects/storefrontlib/src/cms-components/product/product-variants/guards/product-variant.guard.ts]: We have to change it to 3.2 now ;-)

File: feature-libs/product/variants/core/guards/product-variants.guard.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  SemanticPathService,
+  VariantOption,
+} from '@spartacus/core';
+import { Observable, of } from 'rxjs';
+import { filter, map, switchMap, take } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProductVariantsGuard implements CanActivate {
```
Comment (dunqan) [inline feature-libs/product/variants/core/guards/product-variants.guard.ts]: I'm thinking about keeping it in the components entry point (as before) so core won't be basically needed for now?

File: feature-libs/product/variants/assets/translations/en/variants.i18n.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export const variants = {
```
Comment (dunqan) [inline feature-libs/product/variants/assets/translations/en/variants.i18n.ts:1]: We have to deprecate translations that are still in the asset library,

File: feature-libs/product/variants/core/guards/product-variants.guard.ts (hunk 4 above)
Comment (psmul) [reply feature-libs/product/variants/core/guards/product-variants.guard.ts]: Yes, basically (for now) Core was created just to handle guard. Not sure if I like the idea of breaking the concept that we have for future libs, but it's up to discussion.

File: feature-libs/product/package.json (hunk 1 above)
Comment (psmul) [reply feature-libs/product/package.json]: Done. 
Although is there any place I should not this change to be added with the next version?

File: feature-libs/product/public_api.ts (hunk 2 above)
Comment (psmul) [reply feature-libs/product/public_api.ts]: Done

File: projects/storefrontlib/src/cms-components/product/product-variants/guards/product-variant.guard.ts (hunk 3 above)
Comment (psmul) [reply projects/storefrontlib/src/cms-components/product/product-variants/guards/product-variant.guard.ts]: Updated

File: feature-libs/product/yarn.lock [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (dunqan) [inline feature-libs/product/yarn.lock]: Do we need this file?

**Review by dunqan — DISMISSED** [review]

**Review by dunqan — DISMISSED** [review]

**Review by dunqan — APPROVED** [review]

**Pull request merged**